models/unet-2d-nuclei-broad/UNet2DNucleiBroad.model.yaml

```diff
@@ -12,8 +12,8 @@ format_version: 0.1.0
 language: python
 framework: pytorch
 
-source: torch_resource.models.unet.UNet2d
-kwargs: {input_channels: 1, output_channels: 1}
+source: torchbio.models.unet.UNet2d
+default_kwargs: {input_channels: 1, output_channels: 1}
```

Comment:
What's the difference between `default_kwargs` and `kwargs`?

Reply:
the idea is the following:
Any spec can provide required kwargs and default kwargs
spec1.yml:
```yaml
source: ...
required_kwargs: ['a']
default_kwargs: {'b': 2, 'c': 3}
```

if a spec is used in another one kwargs should contain all required kwargs and may overwrite defaults
spec2.yml:
```yaml
some_thing:
    spec: spec1.yml
    kwargs: {'b':5}
```

Reply:
I see, that makes sense. It's not fully clear what to do for the optimizer and the training loop function then, 
because both are not really on the spec level / at least that's debatable.
Also, semantically, I think `required_kwargs` and `optional_kwargs` would make more sense then.
**Edit:** Ok, I understand the separation between `source` and `spec` for the kwargs now and I agree.
But I would still prefer `optional_kwargs` over `default_kwargs`.

Reply:
you mean requried and optional, right?
either optional or default is fine with me. 
for optimizer und alles, was nicht selbst sepc ist, finde ich diese bezeichnungen auch gut, da der Consumer immer noch die eigentlich kwargs bestimmt

Reply:
yep, `required` and `optional`. makes more sense than `required` and `defaults`.